src/pages/workspaces/workspace/Data.js

```diff
@@ -353,72 +354,75 @@ const ReferenceDataContent = ({ workspace: { workspace: { namespace, attributes
   ])
 }
 
-const ToolDrawer = ({ openDrawer, onDismiss, onIgvSuccess, selectedEntities }) => {
+const ToolDrawer = ({ workspace, openDrawer, onDismiss: baseOnDismiss, onIgvSuccess, selectedEntities }) => {
   const [toolMode, setToolMode] = useState()
   const entitiesCount = _.keys(selectedEntities).length
   const entitiesType = !!entitiesCount && selectedEntities[_.keys(selectedEntities)[0]].entityType
+  const onDismiss = () => {
+    baseOnDismiss()
+    setToolMode(undefined)
+  }
 
-  useEffect(() => {
-    if (!openDrawer) {
-      setToolMode(undefined)
-    }
-  }, [openDrawer])
   return h(ModalDrawer, {
     openDrawer,
     onDismiss,
     width: 450
   }, [
-    Utils.switchCase(toolMode, [
-      undefined, () => h(Fragment, [
-        h(TitleBar, {
-          title: 'OPEN WITH...',
-          onDismiss
-        }),
-        div({ style: { display: 'flex', flexDirection: 'column', margin: '0 1.5rem' } }, [
-          div({
-            style: {
-              borderRadius: '1rem',
-              border: `1px solid ${colors.dark(0.5)}`,
-              padding: '0.25rem 0.875rem',
-              alignSelf: 'flex-start',
-              fontSize: 12
-            }
-          }, [
-            `${entitiesCount} ${entitiesType}s selected`
-          ]),
-          div({ style: { margin: '1rem 0' } }, [
-            h(ModalToolButton,
-              {
+    h(Fragment, [
+      h(TitleBar, {
+        title: Utils.switchCase(toolMode,
+          ['IGV', () => 'IGV'],
+          ['Workflow', () => 'YOUR WORKFLOWS'],
+          [Utils.DEFAULT, () => 'OPEN WITH...']),
+        onPrevious: toolMode ? (() => setToolMode(undefined)) : undefined,
+        onDismiss
+      }),
+      div({
+        style: {
+          borderRadius: '1rem',
+          border: `1px solid ${colors.dark(0.5)}`,
+          padding: '0.25rem 0.875rem',
+          margin: '-1rem 1.5rem 1.5rem',
+          alignSelf: 'flex-start',
+          fontSize: 12
+        }
+      }, [
+        `${entitiesCount} ${entitiesType}s selected`
+      ]),
+      Utils.switchCase(toolMode, [
```

Comment:
TOL: Would it be cleaner to use one switchCase, where each item returns an object with the title and the content? 
Something like: 
`[ 'IGV', () => ({ 
    title: IGV, 
    content: h(IGVFileSelector, {
          onSuccess: onIgvSuccess,
          selectedEntities
        })
})`

Reply:
Probably...